### PR TITLE
Task/frontend99 add google authorization

### PR DESCRIPTION
**Pull request by so5tmaker**

### CODE REVIEW

**Comment by so5tmaker** [comment]
[Implement Google Authorization](https://trello.com/c/LHPNqaMk/99-implement-google-authorization)

File: backend/core/src/primary-adapters/rest/auth/auth-controller.ts [hunk 1]
```diff
@@ -95,8 +104,10 @@ export class AuthController extends BaseHttpController {
     @inject(CONTAINER_TYPES.RefreshTokenService) private refreshTokenService: RefreshTokenService,
     @inject(CONTAINER_TYPES.ResetPasswordService) private resetPasswordService: ResetPasswordService,
     @inject(CONTAINER_TYPES.AccountVerificationService) private accountVerificationService: AccountVerificationService,
+    @inject(CONTAINER_TYPES.ProfileRepository) private profileRepository: ProfileRepository,
   ) {
     super();
+    this.profileRepository = profileRepository;
```
Comment (foxt451) [inline backend/core/src/primary-adapters/rest/auth/auth-controller.ts]: if you use the construction above (private .... in parameters), then this... = ... is unnecessary. this construction is typescript's syntanctic sugar that already includes this assignment.

File: backend/core/src/primary-adapters/rest/auth/auth-controller.ts [hunk 2]
```diff
@@ -657,4 +668,126 @@ export class AuthController extends BaseHttpController {
...
+      picture,
+    } = await this.userService.getGoogleUser({ id_token, access_token });
+
+    const googleUser = await this.userService.getUserByEmail(email);
+
+    if (!googleUser) {
+      newUser = await this.userService.createUser({
+        email,
+        username,
+        password,
```
Comment (foxt451) [inline backend/core/src/primary-adapters/rest/auth/auth-controller.ts]: I don't think it's a good idea to set password to id, security-wise. I contacted Vitalii and he said the following: "pssword лучше оставлять пустым". Also, he said the following: "И скажи ему пусть бизнес логику в контроллере не пишет. Пусть в сервис переносит"

**Review by foxt451 — CHANGES_REQUESTED** [review]

File: backend/core/src/primary-adapters/rest/auth/auth-controller.ts (hunk 1 above)
Comment (so5tmaker) [reply backend/core/src/primary-adapters/rest/auth/auth-controller.ts]: Done!

File: backend/core/src/primary-adapters/rest/auth/auth-controller.ts (hunk 2 above)
Comment (so5tmaker) [reply backend/core/src/primary-adapters/rest/auth/auth-controller.ts]: Done!

File: backend/core/src/core/user/application/user-service.ts [hunk 3]
```diff
@@ -123,4 +125,29 @@ export class UserService {
...
+      picture,
+    } = await this.getGoogleUser({ id_token, access_token });
+
+    const googleUser = await this.getUserByEmail(email);
+
+    if (!googleUser) {
+      newUser = await this.createUser({
+        email,
+        username,
+        password,
```
Comment (foxt451) [inline backend/core/src/core/user/application/user-service.ts]: Hi. I can't see if you've corrected one of the previous remarks: "I don't think it's a good idea to set password to id, security-wise. I contacted Vitalii and he said the following: "pssword лучше оставлять пустым"."

**Review by foxt451 — CHANGES_REQUESTED** [review]

File: backend/core/src/core/user/application/user-service.ts (hunk 3 above)
Comment (so5tmaker) [reply backend/core/src/core/user/application/user-service.ts]: Done!

**Review by foxt451 — APPROVED** [review]

**Pull request merged**